packages/core/core/src/Asset.js

```diff
@@ -46,63 +59,175 @@ type SerializedOptions = {|
 
 export default class Asset implements IAsset {
   id: string;
-  hash: string;
+  hash: ?string;
   filePath: FilePath;
   type: string;
-  code: string;
   ast: ?AST;
+  map: ?SourceMap;
+  mapKey: ?string;
   dependencies: Map<string, IDependency>;
   connectedFiles: Map<FilePath, File>;
-  output: AssetOutput;
+  isIsolated: boolean;
   outputHash: string;
   env: Environment;
   meta: Meta;
   stats: Stats;
+  content: Blob;
+  contentKey: ?string;
 
   constructor(options: AssetOptions) {
     this.id =
       options.id ||
       md5FromString(
         options.filePath + options.type + JSON.stringify(options.env)
       );
-    this.hash = options.hash || '';
+    this.hash = options.hash;
     this.filePath = options.filePath;
+    this.isIsolated = options.isIsolated == null ? false : options.isIsolated;
     this.type = options.type;
-    this.code = options.code || (options.output ? options.output.code : '');
+    this.content = options.content || '';
+    this.contentKey = options.contentKey;
     this.ast = options.ast || null;
```

Comment:
Should we make ASTs accessed in the same way (via a method)? If we start caching them we'll want to read them from packagers on demand.